### PR TITLE
DolphinQt: Make Windows tab bar scroll buttons opaque

### PR DESCRIPTION
**Pull request by Dentomologist**
Fixes [12623](https://bugs.dolphin-emu.org/issues/12623).

### CODE REVIEW

File: Source/Core/DolphinQt/Settings.cpp [hunk 1]
```diff
@@ -146,6 +153,25 @@ void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
...
     stylesheet_contents.append(QStringLiteral("%1").arg(tooltip_stylesheet));
   }
+#ifdef _WIN32
+  // MSVC has a bug causing QTabBar scroll buttons to be partially transparent when they inherit any
+  // stylesheet (see https://bugreports.qt.io/browse/QTBUG-74187) which is triggered when setting
+  // qApp's stylesheet below. Setting the scroll buttons' color directly fixes the problem.
+
+  // Create a QToolButton that's a child of a QTabBar in case that has different styling than a
+  // plain QToolButton
+  QTabBar* const tab_bar = new QTabBar();
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Settings.cpp:163]:
```suggestion
  auto* const tab_bar = new QTabBar();
```

File: Source/Core/DolphinQt/Settings.cpp [hunk 2]
```diff
@@ -146,6 +153,25 @@ void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
...
+  QToolButton* const tool_button = new QToolButton(tab_bar);
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Settings.cpp:164]:
```suggestion
  auto* const tool_button = new QToolButton(tab_bar);
```

File: Source/Core/DolphinQt/Settings.cpp [hunk 3]
```diff
@@ -146,6 +153,25 @@ void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
...
+
+  // tab_bar's destructor deletes child tool_button
+  const Common::ScopeGuard delete_widgets([&tab_bar]() { delete tab_bar; });
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Settings.cpp:167]: Consider using a unique_ptr for `tab_bar`.

File: Source/Core/DolphinQt/Settings.cpp [hunk 4]
```diff
@@ -146,6 +153,25 @@ void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
...
+
+  const QColor background_color = tool_button->palette().color(QPalette::Button);
+
+  const std::string style_var =
+      fmt::format("QTabBar QToolButton {{background-color: #{:08x}}};", background_color.rgba());
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Settings.cpp:172]:
```suggestion
      fmt::format("QTabBar QToolButton {{ background-color: #{:08x}; }}", background_color.rgba());
```
for readability

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Comment by Dentomologist** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinQt/Settings.cpp [hunk 5]
```diff
@@ -146,6 +153,22 @@ void Settings::SetCurrentUserStyle(const QString& stylesheet_name)
...
+
+  // Create a temporary QToolButton that's a child of a QTabBar in case that has different styling
+  // than a plain QToolButton.
+  const auto tab_bar = std::make_unique<QTabBar>();
+  auto* const tool_button = new QToolButton(tab_bar.get());
+
+  const QRgb background_color = tool_button->palette().color(QPalette::Button).rgba();
+
+  const std::string style_var =
+      fmt::format("QTabBar QToolButton {{ background-color: #{:08x}; }}", background_color);
```
Comment (leoetlino) [inline Source/Core/DolphinQt/Settings.cpp:169]: `#include <fmt/format.h>` (sorry, missed this earlier)

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Review by leoetlino — APPROVED** [review]

**Pull request merged**